simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/component/chargingstation/ChargeProtocolVariableAccessor.java

```diff
@@ -0,0 +1,67 @@
+package com.evbox.everon.ocpp.simulator.station.component.chargingstation;
+
+import com.evbox.everon.ocpp.common.CiString;
+import com.evbox.everon.ocpp.simulator.station.Station;
+import com.evbox.everon.ocpp.simulator.station.StationHardwareData;
+import com.evbox.everon.ocpp.simulator.station.component.variable.SetVariableValidator;
+import com.evbox.everon.ocpp.simulator.station.component.variable.VariableAccessor;
+import com.evbox.everon.ocpp.simulator.station.component.variable.VariableGetter;
+import com.evbox.everon.ocpp.simulator.station.component.variable.VariableSetter;
+import com.evbox.everon.ocpp.v20.message.centralserver.*;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Collections;
+import java.util.Map;
+
+public class ChargeProtocolVariableAccessor extends VariableAccessor {
+
+    public static final String NAME = "ChargeProtocol";
+
+    private final Map<GetVariableDatum.AttributeType, VariableGetter> variableGetters = ImmutableMap.<GetVariableDatum.AttributeType, VariableGetter>builder()
+            .put(GetVariableDatum.AttributeType.ACTUAL, this::getActualValue)
+            .build();
+
+    private final Map<SetVariableDatum.AttributeType, SetVariableValidator> variableValidators = ImmutableMap.<SetVariableDatum.AttributeType, SetVariableValidator>builder()
+            .put(SetVariableDatum.AttributeType.ACTUAL, this::validateActualValue)
+            .build();
+
+    public ChargeProtocolVariableAccessor(Station station) {
+        super(station);
+    }
+
+    @Override
+    public String getVariableName() {
+        return NAME;
+    }
+
+    @Override
+    public Map<GetVariableDatum.AttributeType, VariableGetter> getVariableGetters() {
+        return variableGetters;
+    }
+
+    @Override
+    public Map<SetVariableDatum.AttributeType, VariableSetter> getVariableSetters() {
+        return Collections.emptyMap();
+    }
+
+    @Override
+    public Map<SetVariableDatum.AttributeType, SetVariableValidator> getVariableValidators() {
+        return variableValidators;
+    }
+
+    private SetVariableResult validateActualValue(Component component, Variable variable, SetVariableDatum.AttributeType attributeType, CiString.CiString1000 attributeValue) {
```

Comment:
SetVariableResult contains attributeStatus field which tells if station is able to process attribute value for given component, evse, connector, variable etc. In this particular case variable is immutable, at least for now, so we always return REJECTED

Reply:
but again, I expect method arguments to be validated, so I would simply name it something like: createSetVariableResult()

Reply:
Purpose of a method is not a creation of an object. It actually validates if passed attributeValue is eligible to be used as a new variable value.

Reply:
I misunderstand something here... Because I see that we are creating an instance of SetVariableResult. 
Can you please point to the code where it actually validates? The only `suspicious` line of code that I see is `.withAttributeType(SetVariableResult.AttributeType.fromValue(attributeType.value()))`

Is that validation?

Reply:
You're right, here it's not actually a validation, but I wanted to keep it for consistency with others, where variable would be mutable: https://github.com/evbox/station-simulator/pull/5/files/a43fef86bf25b6291ecca7ab0061e93073c6029c#diff-c4511a9c83b22deb9631ef09e83a228bR61

Also here's how it looks like in last PR: https://github.com/evbox/station-simulator/pull/9/files#diff-a30524d5497ec98f0da24d23fb3a89c3R53

Please see and tell your final opinion.

Reply:
I would suggest to remove **validate** name from the methods where validation does not happen. Just call this method createSetVariableResult(). Its a private method it will not effect your design. In other cases where validation is needed you can implement validate method.